packages/story-editor/src/app/canvas/useCanvasCopyPaste.js

```diff
@@ -103,6 +104,29 @@ function useCanvasGlobalKeys() {
         currentPage
       );
 
+      const newProductElements = elements.filter(
+        ({ type }) => type === ELEMENT_TYPES.PRODUCT
+      );
+
+      if (
+        currentPageProductIds.length >= MAX_PRODUCTS_PER_PAGE ||
+        newProductElements.length + currentPageProductIds.length >
+          MAX_PRODUCTS_PER_PAGE
+      ) {
+        showSnackbar({
+          message: sprintf(
+            /* translators: %d: max number of products. */
+            _n(
+              'Only %d item can be added per page.',
+              'Only %d items can be added per page.',
+              MAX_PRODUCTS_PER_PAGE,
+              'web-stories'
+            ),
+            MAX_PRODUCTS_PER_PAGE
+          ),
+        });
+      }
```

Comment:
Let's return early here to avoid double snackbar notifications:

```suggestion

      return addPastedElements(elements, animations);
      }
```

<img width="482" alt="Screenshot 2022-06-17 at 16 00 39" src="https://user-images.githubusercontent.com/841956/174314224-0b4f26e5-5cc9-438a-b7d9-30433539e459.png">

Reply:
Good catch. Pushed a fix and improved logic.